test/unit/features/test_changes_follower.py

```diff
@@ -460,7 +460,9 @@ def test_start(self):
         try:
             self.prepare_mock_changes(batches=MAX_BATCHES)
             follower = ChangesFollower(self.client, db='db')
-            count = self.runner(follower, _Mode.LISTEN, timeout=5)
+            count = self.runner(
+                follower, _Mode.LISTEN, timeout=5, stop_after=3 * _BATCH_SIZE
```

Comment:
We're changing the stop condition here from time-based to count-based, but the assertion hasn't changed. Should we be stopping after 2 times batch size? Or asserting that we got at least 3 batches instead of 2?

Reply:
Ok, so the issue here is that this PR changes behavior of the tests from "timeout and hope we have at least this number of changes" to "timeout and wait till we don't have new changes anymore" because Jenkins is not kind to python's thread and our assumptions about "at least this number" is often wrong.

The issue with this test is that after timeout it is waiting for `MAX_BATCHES` to pass and essentially hangs. So  `3 * _BATCH_SIZE` here is not really a condition change, but kind of an unlock of main thread after timeout. I moved those 3 batches from `stop_after` to mock changes, so that'd be less subtle.

Not sure about changing counter assertion, to be honest. We define it as "There should be some changes." which should win an award for most vague test of the year, I mean, technically 1  is "some changes". I can update it and make 3 to match the mock batches, but this is not exactly what we are asserting here.

Reply:
I agree, `>0` changes would be a sufficient assertion for this particular test (which is really testing that the "continuous" mode starts and produces changes).
I was confused by the `stop_after` condition in addition to the timeout, but with the latest change it is more clear.